src/lib/y2storage/boot_requirements_strategies/analyzer.rb

```diff
@@ -0,0 +1,199 @@
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage/planned"
+
+module Y2Storage
+  module BootRequirementsStrategies
+    # Auxiliary class that takes information from several sources (current
+    # devicegraph, already planned devices and user input) and provides useful
+    # information (regarding calculation of boot requirements) about the
+    # expected final system.
+    class Analyzer
+      # Constructor
+      #
+      # @param devicegraph     [Devicegraph] starting situation.
+      # @param planned_devices [Array<Planned::Device>] devices that are already planned to be
+      #   added to the starting devicegraph.
+      # @param boot_disk_name  [String, nil] device name of the disk that the system will try to
+      #   boot first. Only useful in some scenarios like legacy boot.
+      def initialize(devicegraph, planned_devices, boot_disk_name)
+        @devicegraph = devicegraph
+        @planned_devices = planned_devices
+        @boot_disk_name = boot_disk_name
+
+        @root_planned_dev = planned_devices.find do |dev|
+          dev.respond_to?(:mount_point) && dev.mount_point == "/"
+        end
+        @root_filesystem = devicegraph.filesystems.find { |fs| fs.mountpoint == "/" }
+      end
+
+      # Disk in which the system will look for the bootloader.
+      #
+      # This is relevant only in some strategies (mainly legacy boot).
+      #
+      # There is no way to query the system for such value, so this method
+      # relies on #boot_disk_name. If that information is not available, the
+      # (first) disk hosting the root filesystem is considered to be the boot
+      # disk.
+      #
+      # If "/" is still not present in the devicegraph or the list of planned
+      # devices, the first disk of the system is used as fallback.
+      #
+      # @return [Disk]
+      def boot_disk
+        return @boot_disk if @boot_disk
+
+        if boot_disk_name
+          @boot_disk = devicegraph.disks.find { |d| d.name == boot_disk_name }
+        end
+
+        @boot_disk ||= boot_disk_from_planned_dev
+        @boot_disk ||= boot_disk_from_devicegraph
+        @boot_disk ||= devicegraph.disks.first
+
+        @boot_disk
+      end
+
+      # Whether the root (/) filesystem is going to be in a LVM logical volume
+      #
+      # @return [Boolean] true if the root filesystem is going to be in a LVM
+      #   logical volume.
```

Comment:
it should be mentioned here, that if root is not specified in graph or in planned devices, it returns false.